pkg/vsphere/toolbox/service.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package toolbox
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"sync"
+	"time"
+)
+
+// Service receives and dispatches incoming RPC requests from the vmx
+type Service struct {
+	name     string
+	in       Channel
+	out      Channel
+	handlers map[string]Handler
+	stop     chan struct{}
+	wg       *sync.WaitGroup
+
+	Interval  time.Duration
+	PrimaryIP func() string
+}
+
+// NewService initializes a Service instance
+func NewService(rpcIn Channel, rpcOut Channel) *Service {
+	s := &Service{
+		name:     "toolbox", // Same name used by vmtoolsd
+		in:       NewTraceChannel(rpcIn),
+		out:      NewTraceChannel(rpcOut),
+		handlers: make(map[string]Handler),
+		wg:       new(sync.WaitGroup),
+		stop:     make(chan struct{}, 1),
+
+		Interval:  time.Second,
+		PrimaryIP: DefaultIP,
+	}
+
+	s.RegisterHandler("reset", s.Reset)
```

Comment:
No, the "reset" RPC is for the guest<->vmx channel itself, not related to VM power operations.  You'll see a "reset" request at startup, with cmd/toolbox/main.go for example.